Fix the build with glibc 2.34 and later, where PTHREAD_STACK_MIN is no
longer a preprocessor constant. Backport of boostorg/thread commit
74fb0a26099bc51d717f5f154b37231ce7df3e98, included in Boost 1.73.0.

--- a/boost/thread/pthread/thread_data.hpp
+++ b/boost/thread/pthread/thread_data.hpp
@@ -57,7 +57,7 @@ namespace boost
 #else
           std::size_t page_size = ::sysconf( _SC_PAGESIZE);
 #endif
-#if PTHREAD_STACK_MIN > 0
+#ifdef PTHREAD_STACK_MIN
           if (size<PTHREAD_STACK_MIN) size=PTHREAD_STACK_MIN;
 #endif
           size = ((size+page_size-1)/page_size)*page_size;
